app/controllers/deregister/individual/DeregistrationDateController.scala

```diff
@@ -128,4 +132,9 @@ class DeregistrationDateController @Inject()(config: FrontendAppConfig,
         status
     }
 
+  private def getDate(implicit request: DataRequest[AnyContent]): Future[LocalDate] =
+    subscriptionConnector.getPspApplicationDate(request.user.pspIdOrException).map(LocalDate.parse)
```

Comment:
Wouldn't it be better to do the date parse on the connector? Save duplicated code